mypy/checker.py

```diff
@@ -1720,7 +1720,10 @@ def visit_class_def(self, defn: ClassDef) -> None:
             self.binder = ConditionalTypeBinder()
             with self.binder.top_frame_context():
                 with self.scope.push_class(defn.info):
-                    self.accept(defn.defs)
+                    if not any(refers_to_fullname(decorator,
+                                                  'typing.no_type_check')
+                               for decorator in defn.decorators):
+                        self.accept(defn.defs)
```

Comment:
Type checking populates some information that other parts of the type checker may rely on. So this needs to be more elaborate.

For example, this shouldn't generate errors:

```py
import typing

@typing.no_type_check
class A:
    a = int()

    def f(self) -> None:
        self.b = int()

a = A()
a.a  # Cannot determine type of "a"
a.b  # Cannot determine type of "b"
```

One way to deal with this is to give everything in the class the type `Any`. We'd still no type checking, but it actually wouldn't do much. We use `in_checked_function()` in `mypy.checker` to do this. You can add some check for `no_type_check` decorator there (maybe via a `bool` attribute flag) so that `in_checked_function()` returns False inside the class.